File: 001.Two-Sum/cpp/001.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> indies;
        
        //init map
        for(int i=0; i<nums.size(); ++i){
            indies[nums[i]] = i;
        }
        
        //find the key
        for(int i=0; i< nums.size(); ++i){
            int key = target - nums[i];

            if(indies.count(key) && indies[key] != i) {
                return {i, indies[key]};
            }
        }
        return {};
    }
};

File: 001.Two-Sum/go/001.go

func twoSum(nums []int, target int) []int {

	int_map := make(map[int]int, len(nums))

	for index, value := range nums {

		//find the value
		if find, ok := int_map[target-value]; ok {
			//index is not add, find < index
			return []int{find, index}
		}

		//not find, add the map
		int_map[value] = index
	}
	return nil
}
